### PR TITLE
Refactor `MailerInterface::sendMultiple()`

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️

### CODE REVIEW

File: src/MultipleSendResult.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+declare(strict_types=1);
+
+namespace Yiisoft\Mailer;
+
+use Throwable;
+
+/**
+ * Result of {@see MailerInterface::sendMultiple()}.
+ */
+final class MultipleSendResult
```
Comment (samdark) [inline src/MultipleSendResult.php:12]: I'd think about naming a bit:

- `SendResults`
- `Results`

File: src/MailerInterface.php [hunk 2]
```diff
@@ -60,14 +60,13 @@ public function send(MessageInterface $message): void;
...
-     * corresponding exceptions by {@see MessageInterface::withError()}, then returns them.
+     * sending multiple messages in the same batch.
      *
      * @param MessageInterface[] $messages List of email messages, which should be sent.
      *
-     * @return MessageInterface[] List of fails messages.
+     * @return MultipleSendResult The result object that contains all messages and errors for failed sent messages.
      */
-    public function sendMultiple(array $messages): array;
+    public function sendMultiple(array $messages): MultipleSendResult;
```
Comment (samdark) [inline src/MailerInterface.php:69]: How should I know which result corresponds to which message? Array keys?

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/MailerInterface.php (hunk 2 above)
Comment (vjik) [reply src/MailerInterface.php:69]: You get the same instances of message in `$successMessages` and `$failMessages` arrays.

File: src/MultipleSendResult.php (hunk 1 above)
Comment (vjik) [reply src/MultipleSendResult.php:12]: But this object for `sendMultiple` only. To call it shortler is OK?

Comment (samdark) [reply src/MultipleSendResult.php:12]: Yes. I think `SendResults` is alright.

Comment (vjik) [reply src/MultipleSendResult.php]: Done

**Pull request merged**